compiler/noirc_frontend/src/lexer/lexer.rs

```diff
@@ -1087,4 +1087,211 @@ mod tests {
+                ",
+                        s
+                    ),
+                    format!(
+                        "
+                    /*\n{}\n*/
+                ",
+                        s
+                    ),
+                ],
+            ),
+        ]
+    }
+
+    #[test]
+    fn test_big_list_of_naughty_strings() {
+        use std::mem::discriminant;
+
+        let blns_contents = include_str!(env!("BLNS_JSON_PATH"));
+        let blns_base64: Vec<String> =
+            serde_json::from_str(blns_contents).expect("BLNS json invalid");
+        blns_base64.into_iter().for_each(|blns_base64_str| {
+            blns_base64_to_statements(blns_base64_str).into_iter().for_each(
+                |(token_discriminator_opt, blns_program_strs)| {
+                    blns_program_strs.into_iter().for_each(|blns_program_str| {
+                        let mut expected_token_found = false;
+                        let mut lexer = Lexer::new(&blns_program_str);
+                        let mut result_tokens = Vec::new();
+                        loop {
+                            match lexer.next_token() {
+                                Ok(next_token) => {
+                                    result_tokens.push(next_token.clone());
+                                    expected_token_found |= token_discriminator_opt
+                                        .as_ref()
+                                        .map(|token_discriminator| {
+                                            discriminant(token_discriminator)
+                                                == discriminant(&next_token.token())
+                                        })
+                                        .unwrap_or(true);
+
+                                    if next_token == Token::EOF {
+                                        assert!(lexer.done, "lexer not done when EOF emitted!");
+                                        break;
+                                    }
+                                }
+
+                                Err(LexerErrorKind::InvalidIntegerLiteral { .. })
+                                | Err(LexerErrorKind::UnexpectedCharacter { .. })
+                                | Err(LexerErrorKind::UnterminatedBlockComment { .. }) => {
+                                    expected_token_found = true;
+                                }
+                                Err(err) => {
+                                    panic!("Unexpected lexer error found: {:?}", err)
+                                }
+                            }
+                        }
+
+                        assert!(
+                            expected_token_found,
+                            "{}",
+                            format!(
+                                "expected token not found: {:?}\noutput:\n{:?}",
+                                token_discriminator_opt, result_tokens
+                            )
+                        );
```

Comment:
assert should already support format strings:
```suggestion
                        assert!(
                            expected_token_found,
                            "expected token not found: {token_discriminator_opt:?}\noutput:\n{result_tokens:?}",
                        );
```